### PR TITLE
Fix #312

### LINKED ISSUE #312: Incorrect JSON rendering of named primitive types
Named primitive type, when used as a nested or member type is rendered as its type name instead of its value.

For example, the following blueprint:

```apib
# Rendering Test
## A [/a]
### Retrieve [GET]
- response 200 (application/json)
    - attributes
        - a (array[A Type])

### Data Structures

### A Type (string)
#### Sample
Hello
```

renders the response JSON as:

```json
{
  "a": [
    "A Type"
  ]
}
```

the correct behavior should be:

```json
{
  "a": [
    "Hello"
  ]
}
```

### PR DESCRIPTION
**Pull request by klokane**
Incorrect JSON rendering of named primitive types

### CODE REVIEW

File: src/RefractDataStructure.cc [hunk 1]
```diff
@@ -387,13 +387,18 @@ namespace drafter {
...
-                    }
+                        method = it->symbol.variable ? eSample : eElement;
+                    } 
 
                     const RefractElementFactory& f = FactoryFromType(typeName);
-                    types.push_back(f.Create(it->symbol.literal, it->symbol.variable));
+                    types.push_back(f.Create(it->symbol.literal, method));
+
+                    if (!it->symbol.literal.empty()) {
+                    }
```
Comment (pksunkara) [inline src/RefractDataStructure.cc]: Empty if condition?

**Comment by pksunkara** [comment]
@klokane Just one comment. Otherwise, looks good.

File: src/RefractDataStructure.cc (hunk 1 above)
Comment (klokane) [inline src/RefractDataStructure.cc]: sorry, it is rest form testing implementation, will remove

**Comment by pksunkara** [comment]
Will merge once tests pass.

**Pull request merged**